plugins/auto-sizes/hooks.php

```diff
@@ -139,79 +139,142 @@ function auto_sizes_get_width( string $layout_width, int $image_width ): string
+ *                        Defaults to `current_filter()` when not set.
+ * @return string The HTML content.
  */
-function auto_sizes_filter_image_tag( string $content, array $parsed_block ): string {
+function auto_sizes_prime_attachment_caches( string $content, string $context = null ): string {
+	if ( null === $context ) {
+		$context = current_filter();
+	}
+
 	$processor = new WP_HTML_Tag_Processor( $content );
-	$has_image = $processor->next_tag( array( 'tag_name' => 'img' ) );
 
-	// Only update the markup if an image is found.
-	if ( $has_image ) {
-		$processor->set_attribute( 'data-needs-sizes-update', true );
-		if ( isset( $parsed_block['attrs']['align'] ) ) {
-			$processor->set_attribute( 'data-align', $parsed_block['attrs']['align'] );
+	$images = array();
+	while ( $processor->next_tag( array( 'tag_name' => 'IMG' ) ) ) {
+		$class = $processor->get_attribute( 'class' );
+
+		// Only apply the dominant color to images that have a src attribute.
+		if ( ! is_string( $class ) ) {
+			continue;
 		}
 
-		// Resize image width.
-		if ( isset( $parsed_block['attrs']['width'] ) ) {
-			$processor->set_attribute( 'data-resize-width', $parsed_block['attrs']['width'] );
+		if ( preg_match( '/wp-image-([0-9]+)/i', $class, $class_id ) === 1 ) {
+			$attachment_id = absint( $class_id[1] );
+			if ( $attachment_id > 0 ) {
+				$images[] = $attachment_id;
+			}
 		}
+	}
 
-		$content = $processor->get_updated_html();
+	// Reduce the array to unique attachment IDs.
+	$attachment_ids = array_unique( array_filter( $images ) );
+
+	if ( count( $attachment_ids ) > 1 ) {
+		/*
+		 * Warm the object cache with post and meta information for all found
+		 * images to avoid making individual database calls.
+		 */
+		_prime_post_caches( $attachment_ids, false, true );
 	}
+
 	return $content;
 }
-add_filter( 'render_block_core/image', 'auto_sizes_filter_image_tag', 10, 2 );
-add_filter( 'render_block_core/cover', 'auto_sizes_filter_image_tag', 10, 2 );
+add_filter( 'the_content', 'auto_sizes_prime_attachment_caches', 6 );
 
 /**
  * Filter the sizes attribute for images to improve the default calculation.
  *
  * @since 1.1.0
  *
- * @param string $content The block content about to be rendered.
+ * @param string                                                   $content      The block content about to be rendered.
+ * @param array{ attrs?: array{ align?: string, width?: string } } $parsed_block The parsed block.
+ * @param WP_Block                                                 $block        Block instance.
  * @return string The updated block content.
  */
-function auto_sizes_improve_image_sizes_attributes( string $content ): string {
+function auto_sizes_filter_image_tag( string $content, array $parsed_block, WP_Block $block ): string {
 	$processor = new WP_HTML_Tag_Processor( $content );
-	if ( ! $processor->next_tag( array( 'tag_name' => 'img' ) ) ) {
-		return $content;
-	}
+	$has_image = $processor->next_tag( array( 'tag_name' => 'img' ) );
```

Comment:
For consistency with above.

```suggestion
	$has_image = $processor->next_tag( array( 'tag_name' => 'IMG' ) );
```